dcos/httpclient.go

```diff
@@ -0,0 +1,69 @@
+package dcos
+
+import (
+	"net"
+	"net/http"
+	"time"
+)
+
+const (
+	userAgent = ClientName + " - " + Version
+)
+
+type DCOSTransport struct {
+	Config *Config
+	Base   http.RoundTripper
+}
+
+func (t *DCOSTransport) base() http.RoundTripper {
+	if t.Base != nil {
+		return t.Base
+	}
+	return http.DefaultTransport
+}
+
+func (t *DCOSTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	res, err := t.base().RoundTrip(req)
+
+	req.Header.Set("User-Agent", userAgent)
+
+	return res, err
+}
+
+func NewHTTPClient(config *Config) *http.Client {
+	client := http.DefaultClient
+	client.Transport = &http.Transport{
+
+		// Allow http_proxy, https_proxy, and no_proxy.
+		Proxy: http.ProxyFromEnvironment,
+
+		// Set a 10 seconds timeout for the connection to be established.
+		DialContext: (&net.Dialer{
+			Timeout: 10 * time.Second,
```

Comment:
Maybe we should start using constants right away, so others could reuse them? `dcos.DialTimeout`, `dcos.TLShandshakeTimeout`, etc.

Reply:
nice like this proposal will do so